File: backend/src/main/java/com/example/app/domain/entity/Notice.java
package com.example.app.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notice {
    private Long id;
    private String title;
    private String content;
    private Long authorId;
    private String authorName;
    private boolean isPinned;
    private int viewCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/notice.ts
export interface NoticeListItem {
  id: number
  title: string
  authorName: string
  isPinned: boolean
  viewCount: number
  createdAt: string
}

export interface Notice {
  id: number
  title: string
  content: string
  authorId: number
  authorName: string
  isPinned: boolean
  viewCount: number
  createdAt: string
  updatedAt: string
}

export interface NoticeListResponse {
  list: NoticeListItem[]
  total: number
  page: number
  size: number
}

export interface NoticeRequest {
  title: string
  content: string
  isPinned: boolean
}
